tests/test_calculate_random_infections.py

```diff
@@ -78,40 +78,40 @@ def test_random_contact_immune_and_people_without_contacts_are_not_infected():
 
     virus_strain = np.array([0, -1, -1, -1, -1])
     contagiousness_factor = np.array([1])
+    persistency_factor = np.array([1])
 
     (
         newly_infected,
         infection_counter,
-        immune,
         missed,
         was_infected,
     ) = _calculate_infections_by_random_contacts(
         random_contacts,
         infectious,
-        immune,
+        immunity_level,
         virus_strain,
         group_codes,
         assortative_matching_cum_probs,
         indexers,
         susceptibility_factor,
         contagiousness_factor,
+        persistency_factor,
         infection_counter,
         0,
     )
 
     assert (newly_infected == [-1, -1, -1, -1, -1]).all()
     assert (infection_counter == [0, 0, 0, 0, 0]).all()
-    assert (immune == [True, True, False, True, False]).all()
-    assert (missed[[0, 1, 3], :] > 0).all()
-    assert (missed[[2, 4]] == 0).all()
+    assert (missed[[0, 1], :] > 0).all()
+    assert (missed[[2, 3, 4]] == 0).all()
```

Comment:
Questions: Is this an implication of the new code or a sign of some bug?

Reply:
@tobiasraabe

Reply:
Definitely a bug. The extended docstring should clarify the purpose of the tests. The problem likely happens in the loop code for random contacts.